week_6/multi-step-form-ngrx/src/app/components/input-text-field/input-text-field.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit } from '@angular/core';
+import { Store } from '@ngrx/store';
+import {
+  FormArray,
+  FormControl,
+  ReactiveFormsModule,
+  ValidationErrors,
+  FormBuilder,
+  Validators,
+  ValidatorFn,
+} from '@angular/forms';
+import { PersonalInfoState } from '../../state/reducers/personal-info.reducer';
+import { updatePersonalInfo } from '../../state/actions/personal-info.action';
+
+@Component({
+  selector: 'app-input-text-field',
+  standalone: true,
+  imports: [ReactiveFormsModule],
+  templateUrl: './input-text-field.component.html',
+  styleUrls: ['./input-text-field.component.css'],
+})
+export class InputTextFieldComponent implements OnInit {
+  inputFields = [
+    {
+      label: 'Name',
+      key: 'name',
+      type: 'text',
+      placeholder: 'e.g. Stephen King',
+    },
+    {
+      label: 'Email Address',
+      key: 'email',
+      type: 'email',
+      placeholder: 'e.g. stephenking@lorem.com',
+    },
+    {
+      label: 'Phone Number',
+      key: 'number',
+      type: 'text',
+      placeholder: 'e.g. +1 234 567 890',
+    },
+  ];
+  formArray!: FormArray;
+  errors!: ValidationErrors | null;
+
+  constructor(
+    private fb: FormBuilder,
+    private store: Store<{ personalInfo: PersonalInfoState }>
```

Comment:
create a variable for this object and pass that here

Reply:
Done